### PR TITLE
Allow global unit lookups from scripts

### PR DESCRIPTION
**Pull request by mekolat**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [X] I have followed [proper Hercules code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

I got asked numerous times to add support in `buildin_getunits` to search on the whole server, so here goes.

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

<br><br>

---

This PR makes the following buildins obsolete:
- `maprespawnguildid()`
- `playbgmall()`

Should they be deprecated?

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1851)

**Comment by mekolat** [comment]
stupid wafflebot, I hate you

**Comment by mekolat** [comment]
note to self: `BL_MOB` not working; maybe they're not added to iddb

**Comment by AnnieRuru** [comment]
it seems we have the same idea
I also had this idea some time ago that a script command only do for SELF
then another globally used script command that can run for ALL/MAP/AREA

example like announce, ... the bc_ flag is dropped
but another script command getunit( announce("blah"), ALL_CLIENT ) is equal to bc_all
so this will make specialeffect, playbgm, recovery, ... and so on become easier to deal with

just wonder how this PR direction going ?

**Comment by mekolat** [comment]
@AnnieRuru it's ready now, sorry for the delay

I have added a warning about `dynamic_mobs`.

File: doc/script_commands.txt [hunk 1]
```diff
@@ -10007,4 +10013,4 @@ the available flags are:
 	P_AIRSHIP_ITEM_NOT_ENOUGH
 	P_AIRSHIP_ITEM_INVALID
 
----------------------------------------
\ No newline at end of file
+---------------------------------------
```
Comment (mekolat) [inline doc/script_commands.txt]: there was a missing newline at the end of the file

**Comment by mekolat** [comment]
> @AnnieRuru <br>
but another script command `getunit( announce("blah"), ALL_CLIENT )` is equal to bc_all


Because of script engine limitations this wouldn't be possible (we can't pass buildins to other buildins without evaluating them), so we'd have to come up with a new syntax, such as

```c
// in a square area:
foreach (BL_PC) in ("map", x, y) do command(arg1, arg2, …);

// in a whole map:
foreach (BL_PC) in "map" do command(arg1, arg2, …);

// everywhere:
foreach (BL_PC) do command(arg1, arg2, …);

// combined search:
foreach (BL_PC | BL_MOB) in ("map", x, y) do command(arg1, arg2, …);
```

**Comment by AnnieRuru** [comment]
>so we'd have to come up with a new syntax, such as

exactly my thought ...
the way you do it reminds me from Haru suggestion ...
try do forum search "foreach" on the forum in Station S > Developers Area

**Pull request merged**